### DISCUSSION TITLE
Problem with Section 4 Lesson 8: Chainlink foundry remapping not working

### DISCUSSION THREAD

**Discussion by Play3rZer0** [Cyfrin/foundry-full-course-cu#2771] [Q&A] [answered]
I have tried the suggested solutions on this topic, but none worked.

The problem is that the Solidity compiler cannot find the mapping to contract VRFConsumerBaseV2Plus.sol.

Here is my line of code (in Raffle.sol file):

```

import {VRFConsumerBaseV2Plus} from "@chainlink/contracts/src/v0.8/vrf/dev/VRFConsumerBaseV2Plus.sol";

```

In my foundry.toml file I have included the remapping:

```

remappings = [
    "@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/",
]

```

I also have the smartcontractkit installed in the lib folder. I also verified that the path exists to the contract file VRFConsumerBaseV2Plus.sol. The problem goes away when I use the absolute path:

"lib/chainlink-brownie-contracts/contracts/src/v0.8/vrf/dev/"

However I get errors when compiling: (here are some excerpts):

Error (7576): Undeclared identifier.
  --> src/Raffle.sol:90:40:
   |
90 |                         nativePayment: enableNativePayment
   |                                        ^^^^^^^^^^^^^^^^^^^

What could be the problem and do I need to reinstall everything from scratch? I used version 1.1.1 of the chainlink contracts and even tried 1.1.3 but still did not work when remapping. 

![VRFConsumerBaseV2Plus sol - projects (Workspace) - Visual Studio Code 10_29_2024 3_16_52 PM](https://github.com/user-attachments/assets/3dcbe4dd-dbe2-4214-a1ee-a2a6c657d995)

**Comment by EngrPips** [comment]
Hello @Play3rZer0. If remapping is giving you a headache please use the relative path of the file to make your import.

**Reply by Play3rZer0** [reply, marked as answer]
I found the solution. I just restarted VS Code, and suddenly the remapping worked! As developers we sometimes look for the least simplest solution, and that was probably the first thing to do when there is something weird going on.

**Reply by EngrPips** [reply]
Programming is indeed fun.
